load-test-framework/src/main/java/com/google/pubsub/flic/processing/MessageProcessingHandler.java

```diff
@@ -140,14 +140,16 @@ public void printStats(long start, DelayTrackingThreadPool executor, AtomicBoole
                 latencyStats.getValueAtPercentile(50),
                 latencyStats.getValueAtPercentile(95),
                 latencyStats.getValueAtPercentile(99)));
-    log.info(
-        "The average delay for processing "
-            + latencyType
-            + " latency was "
-            + executor.getAverageTaskWaitTime()
-            + ""
-            + " "
-            + "ms");
+    if(executor != null) {
```

Comment:
Passing in null was a temporary fix because the Kafka publishing task code wasn't tracking the tasks in the same way, more relying on the Apache client to do the work for it. Because it wasn't updating the executor or using it at all, passing in null was easier. After thinking about it more, I realized that the way it's set up now doesn't give a 1:1 comparison. On the CPS side, the publisher tracks the time up until when the batch is sent off, but the Kafka publishing code doesn't do this, it just sets the start when it first starts sending any messages, so that batch->send delay is built-in against Kafka. With a large load test, that delay could be pretty significant, so I think it's best for there to be no data rather than bad data and hold off on putting this in.

I'll work on those changes and let you know when this PR is ready again.